lib/devise/strategies/passwordless_authenticatable.rb

```diff
@@ -0,0 +1,29 @@
+require "devise/strategies/authenticatable"
+require_relative "../../../app/mailers/user_mailer"
+
+module Devise
+  module Strategies
+    class PasswordlessAuthenticatable < Authenticatable
+      def authenticate!
+        if params[:user].present?
+          user = User.find_by(email: params[:user][:email])
+
+          if user&.update(
+            login_token: SecureRandom.hex(10),
+            login_token_valid_until: 60.minutes.from_now)
+
+            url = Rails.application.routes.url_helpers.email_confirmation_url(
+              login_token: user.login_token,
+              host: Rails.application.config.domain,
+            )
+
+            UserMailer.sign_in_email(user, url).deliver_now
```

Comment:
Maybe `deliver_later` here?

Reply:
Deliver_later would be a bad idea especially when the system gets big and busy. User could be waiting 5 minutes for their login email which would be not so good.